Sming/sming.mk

```diff
@@ -0,0 +1,662 @@
+	#   (RBOOT_BIG_FLASH seems to be required even for 1M devices)
+	RBOOT_RTC_ENABLED   ?= 0
+	RBOOT_GPIO_ENABLED  ?= 0
+	RBOOT_BIG_FLASH     ?= 1
+
+	ifeq ($(SPI_SIZE),1M)
+		RBOOT_TWO_ROMS  	?= 1
+	else
+		RBOOT_TWO_ROMS      ?= 0
+	endif
+
+	#   place upper spiffs below user flash area
+	RBOOT_SPIFFS_1  ?= $(call hexcalc, $(USER_FLASH_ADDR) - $(SPIFF_SIZE))
+
+	#   place lower spiffs (we loose a few sectors to simplify things)
+	RBOOT_SPIFFS_0  ?= $(call hexcalc, $(RBOOT_SPIFFS_1) - $(MEM_SIZE) / 2)
+
+	#   generate linker files (Sming needs 0x2010 instead of 0x2000!)
+	ROM0_ADDR       := 0x002010
+	IROM0_SIZE      ?= $(call hexcalc, 0x100000 - $(ROM0_ADDR))
+	ROM0_LD         ?= $(BUILD_BASE)/rom0.ld
+
+	#   rom files
+	RBOOT_BIN       := $(FW_BASE)/rboot.bin
+	ROM0_BIN        := $(FW_BASE)/rom0.bin
+
+	#	special settings for two-rom mode
+	ifeq ($(RBOOT_TWO_ROMS),1)
+		ROM1_BIN        := $(FW_BASE)/rom1.bin
+		ROM1_LD         ?= $(BUILD_BASE)/rom1.ld
+
+		#   this is where rboot expects the 2nd rom in two-rom mode
+		ROM1_ADDR       ?= $(call hexcalc, $(ROM0_ADDR) + $(MEM_SIZE) / 2)
+
+		#	this seems to be a save value (found by trial and error)
+		IROM0_SIZE      := $(call hexcalc, $(RBOOT_SPIFFS_0) - 0x8000)
+	endif
+
+	OTA_FILES       += $(ROM0_BIN) $(ROM1_BIN)
+	FLASH_FILES     += $(RBOOT_BIN) $(ROM0_BIN)
+	FLASH_ROM_FLAGS += 0x000000 $(RBOOT_BIN) 0x002000 $(ROM0_BIN)
+
+	#   find rboot source files
+	MODULES         += $(THIRD_PARTY_DIR)/rboot/appcode
+
+	DEFINES += RBOOT_INTEGRATION  RBOOT_RTC_ENABLED RBOOT_GPIO_ENABLED
+	DEFINES += RBOOT_TWO_ROMS BOOT_BIG_FLASH RBOOT_SPIFFS_0 RBOOT_SPIFFS_1
+
+#==============================================================================
+#   non-rboot settings ("standalone applications")
+#------------------------------------------------------------------------------
+else
+	#   place spiffs below user flash area
+	RBOOT_SPIFFS_0  ?= $(call hexcalc, $(USER_FLASH_ADDR) - $(SPIFF_SIZE))
+
+	#	generate linker files
+	ROM0_ADDR       ?= 0x00A000
+	IROM0_SIZE      ?= $(call hexcalc, 0x100000 - $(ROM0_ADDR))
+	ROM0_LD         ?= $(BUILD_BASE)/rom0.ld
+
+	#	rom files
+	IMAGE_MAIN      := $(FW_BASE)/main.bin
+	IMAGE_SDK       := $(FW_BASE)/sdk.bin
+	FLASH_FILES     += $(IMAGE_MAIN) $(IMAGE_SDK)
+	FLASH_ROM_FLAGS += 0x000000 $(IMAGE_MAIN) $(ROM0_ADDR) $(IMAGE_SDK)
+endif
+
+#==============================================================================
+#   show memory map
+#------------------------------------------------------------------------------
+ifeq ($(V),1)
+    $(info )
+    $(info FLASH MEMORY MAP)
+    $(info )
+    $(info MEM_SIZE        = $(MEM_SIZE))
+    $(info ESP_PARM_ADDR   = $(ESP_PARM_ADDR))
+    $(info ESP_INIT_ADDR   = $(ESP_INIT_ADDR))
+    $(info RF_CAL_ADDR     = $(RF_CAL_ADDR))
+    ifneq ($(USER_FLASH_SIZE),0)
+        $(info USER_FLASH_ADDR = $(USER_FLASH_ADDR) ($(USER_FLASH_SIZE) bytes))
+    endif
+    ifneq ($(SPIFF_SIZE),0)
```

Comment:
I'm not sure, whether I understand this. There are calculations in both branches, but those lines are for debug info only.